java/src/main/java/org/rocksdb/AbstractMutableOptions.java

```diff
@@ -199,7 +199,7 @@ protected U setIntArray(
     }
 
     @SuppressWarnings("unchecked")
-    protected <N extends Enum<N>> N getEnum(final K key)
+    protected <N extends Enum<N>> N getEnum(final K key, Class<N> cls)
```

Comment:
Since getEnum isn't doing any type checking on N or cls, wouldn't be more honest to return `Enum<?>`?